content/blog/case-weights/index.Rmd

```diff
@@ -0,0 +1,310 @@
+    weights = n,
+    family = "binomial"
+  )
+glm_fit
+```
+
+
+_This is not quite right though_. There are 12 combinations of `Gender` and `Dept`. How can the model have 23 total degrees of freedom?
+
+If we are treating our data as binomial, the traditional method for fitting this model is to convert the data to a format with columns for the number of events and non-events (per covariate pattern). Let's convert our data into that format:
+
+```{r glm-events}
+ucb_grouped_data <- 
+  ucb %>% 
+  pivot_wider(
+    id_cols = c(Gender, Dept),
+    names_from = Admit,
+    values_from = n,
+    values_fill = 0L
+  )
+ucb_grouped_data
+```
+
+Now, since there are really only `r nrow(ucb_grouped_data)` covariate combinations, the appropriate model can be created. 
+
+
+```{r glm-grouped}
+glm(
+  cbind(Rejected, Admitted) ~ Gender + Dept,
+  data = ucb_grouped_data,
+  family = binomial
+)
+```
+
+In both cases the model coefficients are the same; the standard errors and degrees of freedom are correct for the model with grouped data though. 
+
+## Why is this so complicated? 
+
+Traditionally, the use of weights in base R functions are to fit the model and to report limited measures of model fit. Here, `glm()` reports the deviance while `lm()` shows estimates of the RMSE and adjusted-R<sup>2</sup>.
+
+Believe it or not, the logistic regression code shown above is somewhat straight-forward since the scope of the analysis is very small. There are a few things that we do in modern data analysis that `glm()` and `lm()` do not: 
+
+* Resampling (e.g. the bootstrap or cross-validation).
+* Preprocessing methods such as centering and scaling. 
+* Expanded measures of performance (e.g. area under the ROC curve, mean absolute deviations, Kohen's Kappa, and so on)
+
+A framework like tidymodels enables all of these tools and the impact of case weights is more comprehensive. 
+
+Additionally, the type of case weights **and their intent** affect which of these operations should be affected. 
+
+For example, frequency weights should affect the model fit, the preprocessing steps, and performance estimation. If the predictors require centering, a weighted mean should be used to appropriately ensure that the mean of that column is truly zero. Let's say that sensitivity and specificity estimates are required. The 2x2 table of observed and predicted results should have cell counts that reflect the case weights. If they did not, infrequently occurring data points have as much weight as the rows that have a high prevalence. 
+
+As a counter example, importance weights reflect the idea that they should only influence _the model fitting procedure_. It wouldn't make sense to use a weighted mean to center a predictor; the weight shouldn't influence an unsupervised operation in the same way as the model fit. More critically, any holdout data set used to quantify model efficacy should reflect the data as seen in the wild (without the impact of the weights). 
```

Comment:
The current implementation of `step_center()` uses case weights no matter the type.